src/overlays/effects/ovl_Effect_Ss_Lightning/z_eff_ss_lightning.c

```diff
@@ -1,29 +1,157 @@
+    this->rNumBolts = initParams->numBolts;
+    this->rScale = initParams->scale;
+    this->rYaw = initParams->yaw;
+    this->rLifespan = initParams->life;
+
+    if (!sIsDesegmented) {
+        for (i = 0; i < ARRAY_COUNT(sLightningTextures); i++) {
+            sLightningTextures[i] = Lib_SegmentedToVirtual(sLightningTextures[i]);
+        }
+        sIsDesegmented = true;
+    }
+
+    return 1;
+}
+
+void EffectSsLightning_NewLightning(GlobalContext* globalCtx, Vec3f* pos, s32 yaw, EffectSs* this) {
+    EffectSs newLightning;
+
+    EffectSS_Delete(&newLightning);
+    newLightning = *this;
+    newLightning.pos = *pos;
+    newLightning.rNumBolts--;
+    newLightning.rYaw = yaw;
+    newLightning.life = newLightning.rLifespan;
+
+    EffectSS_Copy(globalCtx, &newLightning);
+}
+
+void EffectSsLightning_Draw(GlobalContext* globalCtx, u32 index, EffectSs* this) {
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    MtxF mfResult;
+    MtxF mfTrans;
+    MtxF mfScale;
+    MtxF mfRot;
+    MtxF mfTransBillboard;
+    MtxF mfTransBillboardRot;
+    Mtx* mtx;
+    f32 yScale;
+    s32 texIndex;
+    f32 xzScale;
+
+    OPEN_DISPS(gfxCtx);
+
+    yScale = this->rScale * 0.01f;
+    texIndex = this->rLifespan - this->life;
+
+    if (texIndex >= ARRAY_COUNT(sLightningTextures)) {
+        texIndex = ARRAY_COUNT(sLightningTextures) - 1;
+    }
+
+    SkinMatrix_SetTranslate(&mfTrans, this->pos.x, this->pos.y, this->pos.z);
+    xzScale = yScale * 0.6f;
+    SkinMatrix_SetScale(&mfScale, xzScale, yScale, xzScale);
+    SkinMatrix_SetRotateRPY(&mfRot, this->vec.x, this->vec.y, this->rYaw);
+    SkinMatrix_MtxFMtxFMult(&mfTrans, &globalCtx->billboardMtxF, &mfTransBillboard);
+    SkinMatrix_MtxFMtxFMult(&mfTransBillboard, &mfRot, &mfTransBillboardRot);
+    SkinMatrix_MtxFMtxFMult(&mfTransBillboardRot, &mfScale, &mfResult);
+
+    gSPMatrix(POLY_XLU_DISP++, &gIdentityMtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    mtx = SkinMatrix_MtxFToNewMtx(gfxCtx, &mfResult);
+
+    if (mtx != NULL) {
+        gSPMatrix(POLY_XLU_DISP++, mtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        func_8012C9BC(gfxCtx);
+        gSPSegment(POLY_XLU_DISP++, 0x08, sLightningTextures[texIndex]);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, this->rPrimColorR, this->rPrimColorG, this->rPrimColorB,
+                        this->rPrimColorA);
+        gDPSetEnvColor(POLY_XLU_DISP++, this->rEnvColorR, this->rEnvColorG, this->rEnvColorB, this->rEnvColorA);
+        gSPDisplayList(POLY_XLU_DISP++, this->gfx);
+    }
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void EffectSsLightning_Update(GlobalContext* globalCtx, u32 index, EffectSs* this) {
+    s32 pad;
+    Vec3f pos;
+    s32 yaw;
+    f32 scale;
+    s32 sign;
+
+    if ((this->rNumBolts != 0) && ((this->life + 1) == this->rLifespan)) {
+        sign = (Rand_ZeroOne() < 0.5f) ? -1 : 1;
+        yaw = this->rYaw + sign * ((s32)((Rand_ZeroOne() * 3640.0f)) + 0xE38);
```

Comment:
```suggestion
        yaw = this->rYaw + sign * ((s32)((Rand_ZeroOne() * 3640.0f)) + 3640);
```

Reply:
This is an angle, so for now it goes in hex. We'd write both in hex if we had support for hex floats, but unfortunately that's C99. It's really 20 degrees, it seems (0x10000 * 20 / 360).

Reply:
Ah that's right, it is an angle. Although it's a bit unfortunate, since it hides the fact that both values are the same.

I wonder if that degree function would work as a macro (outside the scope of this pr ofc)